third_party/move/move-core/types/src/resolver.rs

```diff
@@ -59,18 +71,28 @@ pub trait MoveResolver: ModuleResolver + ResourceResolver {
             .get_resource_with_metadata(address, typ, &self.get_module_metadata(&typ.module_id()))?
             .0)
     }
+
+    fn get_resource_with_metadata(
+        &self,
+        address: &AccountAddress,
+        typ: &StructTag,
+        metadata: &[Metadata],
+    ) -> Result<(Option<Vec<u8>>, usize), Error> {
+        self.get_resource_with_metadata_and_layout(address, typ, metadata, None)
+    }
 }
 
 impl<T: ModuleResolver + ResourceResolver + ?Sized> MoveResolver for T {}
 
 impl<T: ResourceResolver + ?Sized> ResourceResolver for &T {
```

Comment:
What happens if we implement it for T?

Reply:
Hmm, so I don't think we need these impls at all. From my understanding usually you want to have &T is to avoid calling the trait on self, i.e. consuming owned value. For T it should also be fine because we don't have any methods which consume itself. Let's get rid of these maybe?